### PR TITLE
replace flock with fcntl for better os compatibility

### PR DESCRIPTION
**Pull request by sajattack**

### CODE REVIEW

File: src/openrct2/platform/posix.c [hunk 1]
```diff
@@ -282,10 +282,22 @@ bool platform_lock_single_instance()
         log_warning("Cannot open lock file for writing.");
         return false;
     }
-    if (flock(pidFile, LOCK_EX | LOCK_NB) == -1) {
+    #define   LOCK_SH   1    /* shared lock */
```
Comment (Gymnasiast) [inline src/openrct2/platform/posix.c]: You define constants here you never use.

File: src/openrct2/platform/posix.c [hunk 2]
```diff
@@ -282,10 +282,22 @@ bool platform_lock_single_instance()
...
+    #define   LOCK_EX   2    /* exclusive lock */
+    #define   LOCK_NB   4    /* don't block when locking */
+    #define   LOCK_UN   8    /* unlock */
+    
+    struct flock lock;
```
Comment (Gymnasiast) [inline src/openrct2/platform/posix.c]: ?

File: src/openrct2/platform/posix.c [hunk 3]
```diff
@@ -282,10 +282,22 @@ bool platform_lock_single_instance()
...
+    lock.l_start = 0;
+    lock.l_len = 0;
+    lock.l_type = F_WRLCK;
+    lock.l_whence = SEEK_SET;
+
+    if (fcntl(pidFile, F_SETLK, &lock) == -1) {
         if (errno == EWOULDBLOCK) {
             log_warning("Another OpenRCT2 session has been found running.");
-            return false;
+                return false;
```
Comment (Gymnasiast) [inline src/openrct2/platform/posix.c]: Incorrect indentation.

File: src/openrct2/platform/posix.c (hunk 2 above)
Comment (IntelOrca) [reply src/openrct2/platform/posix.c]: In C, `struct` is required before definitions unless typedefed.

File: src/openrct2/platform/posix.c [hunk 4]
```diff
@@ -282,10 +282,22 @@ bool platform_lock_single_instance()
...
+    #define   LOCK_UN   8    /* unlock */
+    
+    struct flock lock;
+
+    lock.l_start = 0;
+    lock.l_len = 0;
+    lock.l_type = F_WRLCK;
+    lock.l_whence = SEEK_SET;
+
+    if (fcntl(pidFile, F_SETLK, &lock) == -1) {
```
Comment (Gymnasiast) [inline src/openrct2/platform/posix.c]: Could you put the brace on its own line?

**Pull request merged**